src/scripting/lua_ui.cc

```diff
@@ -920,513 +1120,1270 @@ static std::function<void()> create_plugin_action_lambda(lua_State* L, const std
 // static, recursive function that does all the work for create_child()
 UI::Panel* LuaPanel::do_create_child(lua_State* L, UI::Panel* parent, UI::Box* as_box) {
 	luaL_checktype(L, -1, LUA_TTABLE);
-
-	// Read some common properties
 	std::string widget_type = get_table_string(L, "widget", true);
-	std::string tooltip = get_table_string(L, "tooltip", false);
-	int32_t x = get_table_int(L, "x", false);
-	int32_t y = get_table_int(L, "y", false);
-	int32_t w = get_table_int(L, "w", false);
-	int32_t h = get_table_int(L, "h", false);
```

Comment:
You can read these properties into a struct here and pass it to the create functions to remove the duplicate reads.